packages/chevrotain/src/parse/grammar/checks.ts

```diff
@@ -53,10 +37,13 @@ import {
   IGrammarValidatorErrorMessageProvider,
   IParserDefinitionError
 } from "./types"
-import dropRight from "lodash/dropRight"
-import compact from "lodash/compact"
 import { tokenStructuredMatcher } from "../../scan/tokens"
 
+// ES2019 Array.prototype.flatMap
+function flatMap<U, R>(arr: U[], callback: (x: U, idx: number) => R[]): R[] {
```

Comment:
TODO: extract to `utils` / `functional-utils` file

Reply:
Or maybe change the compilation target to ES2019?

However, compilation target changes should be done in a separate PR, so the extraction to utils should still be done...